x-pack/plugins/security_solution/public/resolver/view/resolver_loading_state.test.tsx

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Simulator } from '../test_utilities/simulator';
+import { noAncestorsTwoChildren } from '../data_access_layer/mocks/no_ancestors_two_children';
+import { noResolverDataReturned } from '../data_access_layer/mocks/no_resolver_data_returned';
+import '../test_utilities/extend_jest';
+
+describe('Resolver: data loading and resolution states', () => {
+  let simulator: Simulator;
+  let dbDocumentId: string;
+  const resolverComponentInstanceID = 'resolver-loading-state';
+  const newTreeId = 'new-tree-id';
+
+  describe('When resolver is loaded with data', () => {
+    beforeEach(() => {
+      const {
+        metadata: { databaseDocumentID },
+        dataAccessLayer,
+      } = noAncestorsTwoChildren();
+
+      dbDocumentId = databaseDocumentID;
+      simulator = new Simulator({
+        dataAccessLayer,
+        databaseDocumentID,
+        resolverComponentInstanceID,
+      });
+    });
+
+    it('should display the resolver graph', async () => {
+      await expect(
+        simulator.map(() => ({
+          resolverGraphLoading: simulator.testSubject('resolver:graph:loading').length,
+          resolverGraphError: simulator.testSubject('resolver:graph:error').length,
+          resolverGraph: simulator.processResolverGraph(dbDocumentId).length,
+        }))
+      ).toYieldEqualTo({
+        resolverGraphLoading: 0,
+        resolverGraphError: 0,
+        resolverGraph: 1,
+      });
+    });
+  });
+
+  describe('When resolver data is being requested', () => {
+    beforeEach(() => {
+      const {
```

Comment:
this before each seems like the one right above: 'When resolver is loaded with data'

Reply:
Removed the prior test which was unnecessary given the final test in this file